### DISCUSSION TITLE
Need IME support

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#57] [Minor issue]
IME frameworks like Fcitx can't be activated in the program.

**Comment by KennedyRichard** [comment]
Among the changes in the new version to be published soon, I have updated the keyboard input mechanism to one that allows IME frameworks to work in the app. It is experimental work on the app, though, so it might need fixing or some tweaks once launched, but rest assured I'm on it.

By the way, I upload a copy of my local repository from time to time so people can inspect my work if they want. You can access it on this link if you are curious: https://indiepython.com/local-commits. The commit related to changes that should enable IME input are on the commit that mentions the TEXTINPUT events. I'm just mentioning this in case you are curious, this changes should be available in a week or so, but I can't give an accurate date, because it is a lot of stuff.

I usually update it once every 02 or 03 weeks, but this time it took longer because I had to write a few new chapters for the manual regarding new features that are already implemented in my local copy and will be released in the next version and other missing information.

**Reply by GrimPixel** [reply]
I backed up the old package, replaced it with the new one. Fcitx is still not activated.
Emacs has that problem. There is an [article](https://fcitx-im.org/wiki/Input_method_related_environment_variables) on the Fcitx site.

**Reply by KennedyRichard** [reply]
Thank you for the feedback/taking the time to test this.

The change is not directly related to IME input, but it was indeed supposed to have enabled it, as I mentioned.

I'll schedule time to solve this after I'm finished with my current list of work. Unfortunately it will take some time until I can tend do this though. I'm estimating at least a month and half, probably more, because after I release the new version in a week or so I still have to

- create a discussion on improving/implementing graph execution algorithms in Nodezator
- then review the current pending pull request
- then I have to work a bit on the other Indie Python project [Bionic Blue](https://github.com/IndiePython/bionic-blue).

I'm sorry for this taking so long. There's just too many things to be done in the project.

However, I'll do a small research on the topic asap, including the link you mentioned. Depending on the complexity of the change I might be able to find a bit of time between my pending tasks to solve this, but I can't make any promises.

Thanks again for your time and feedback.

Peace

**Reply by GrimPixel** [reply]
I think you may open a new branch to conduct the change, instead of uploading to another place.

**Reply by KennedyRichard** [reply]
Oh, yes, I usually do that. The copy of my local repo is only used for a very specific scenario: when I have to make experimental changes that I expect to span several commits and require countless editions/fine-tuning of the changes/commits.

In other words, to avoid publishing long-lived branches whose commits are constantly changing, causing forks of that branch to break, I work locally exclusively in such cases. In such occasions I upload my local work from time to time for the sake of transparency, so people can inspect my work if they want to, specially people who donate to the project.

I have been using this method of working exclusively locally since a few months ago to implement automated GUI tests but paused my work on this feature midway to implement some new features and reduce a bit the list of stuff to implement.

The work regarding automated GUI tests was progressing fine and thankfully I was able to lay down the foundations already and come up with a more or less complete design on paper. All that's left is to implement the rest of the design, something that I plan to do in a few months, after tackling some other urgent development/management tasks. Here's a preview/demonstration of the feature, in case you are curious: https://www.youtube.com/watch?v=UfXhMTXHJHs.

**Reply by KennedyRichard** [reply]
Speaking of which, many of the 100 and few commits to be published in the new version are related to automated GUI testing (I'd say almost half of them). Though the feature isn't ready yet, such commits are a stable foundation on which I've been working on continuously in the past few months before pausing my work on them to implement the new features for the next release (nodes with callable mode, collapsed mode, the new viewer nodes and more).
